mempool/v1/mempool.go

```diff
@@ -185,43 +188,51 @@ func (txmp *TxMempool) TxsAvailable() <-chan struct{} { return txmp.txsAvailable
 // the size of tx, and adds tx instead. If no such transactions exist, tx is
 // discarded.
 func (txmp *TxMempool) CheckTx(tx types.Tx, cb func(*abci.Response), txInfo mempool.TxInfo) error {
+	cachedTx := tx.ToCachedTx()
 	// During the initial phase of CheckTx, we do not need to modify any state.
+	// A transaction will not actually be added to the mempool until it survives
+	// a call to the ABCI CheckTx method and size constraint checks.
+	height, err := func() (int64, error) {
```

Comment:
so checktx is now async?

Reply:
yeah, this PR reverts https://github.com/celestiaorg/celestia-core/pull/1553 and https://github.com/celestiaorg/celestia-core/pull/1582 cause those break CAT and we need a working mempool that doesn't re-gossip everything